account_invoice_payment_report/models/account_invoice.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Camptocamp SA
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl)
+from odoo import models, fields
+from odoo.tools import float_is_zero
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = 'account.invoice'
+
+    # Backport of odoo v12.0 method
+    def _get_payments_vals(self):
+
+        self.ensure_one()
+        if not self.payment_move_line_ids:
+            return []
+        payment_vals = []
+        currency_id = self.currency_id
```

Comment:
`self.currency_id` returned a currency (record set), not the id.
s/currency_id/currency

Reply:
i would just drop that line